src/MSBuild/XMake.cs

```diff
@@ -2765,62 +2765,53 @@ DirectoryGetFiles getFiles
                     }
+                    potentialProjectFiles = actualProjectFiles.ToArray();
                 }
 
+                // Get all files in the current directory that have a sln extension
+                string[] potentialSolutionFiles = getFiles(projectDirectory ?? ".", "*.sln");
+                string[] solutionFilterFiles = Array.Empty<string>();
                 if (potentialSolutionFiles != null)
                 {
+                    List<string> actualSolutionFiles = new List<string>(potentialSolutionFiles.Length);
+                    List<string> actualSolutionFilterFiles = new List<string>(potentialSolutionFiles.Length);
                     foreach (string s in potentialSolutionFiles)
                     {
-                        if (!FileUtilities.IsSolutionFilename(s))
+                        if (!extensionsToIgnore.Contains(Path.GetExtension(s)))
                         {
-                            extensionsToIgnore.Add(Path.GetExtension(s));
+                            if (FileUtilities.IsSolutionFilterFilename(s))
+                            {
+                                actualSolutionFilterFiles.Add(s);
+                            }
+                            else if (FileUtilities.IsSolutionFilename(s))
+                            {
+                                actualSolutionFiles.Add(s);
+                            }
                         }
                     }
+                    potentialSolutionFiles = actualSolutionFiles.ToArray();
+                    solutionFilterFiles = actualSolutionFiles.ToArray();
                 }
 
-                Dictionary<string, object> extensionsToIgnoreDictionary = ValidateExtensions(extensionsToIgnore.ToArray());
-
-                // Remove projects that are in the projectExtensionsToIgnore List
-                // If we have no extensions to ignore we can skip removing any extensions
-                if (extensionsToIgnoreDictionary.Count > 0)
-                {
-                    // No point removing extensions if we have no project files
-                    if (potentialProjectFiles?.Length > 0)
-                    {
-                        potentialProjectFiles = RemoveFilesWithExtensionsToIgnore(potentialProjectFiles, extensionsToIgnoreDictionary);
-                    }
-
-                    // No point removing extensions if we have no solutions
-                    if (potentialSolutionFiles?.Length > 0)
-                    {
-                        potentialSolutionFiles = RemoveFilesWithExtensionsToIgnore(potentialSolutionFiles, extensionsToIgnoreDictionary);
-                    }
-                }
                 // If there is exactly 1 project file and exactly 1 solution file
-                if ((potentialProjectFiles.Length == 1) && (potentialSolutionFiles.Length == 1))
+                if (potentialProjectFiles.Length == 1 && (potentialSolutionFiles.Length == 1 || solutionFilterFiles.Length == 1))
```

Comment:
does this make sense for the filter case?

Reply:
I'm not 100% sure what's right here. This makes sense to me as it provides a convenient way for you to make it automatically notice the .slnf—just don't have a .sln—but it does technically mean a change in behavior from before we started supporting .slnf, since you could have a .slnf in a folder without a .sln and with a .proj, and it would fall to the check if projectFiles.Length > 1.This would also make it a little harder to notice if you have two .sln files in the folder, since 2 .sln with one .slnf would (with this change) not throw an error but build the .slnf.

The second part is especially problematic in my mind. Do you think I should just revert this change? Revert it then add another check later?